datafusion/optimizer/src/optimizer.rs

```diff
@@ -419,6 +412,34 @@ impl Optimizer {
     }
 }
 
+/// Returns an error if plans have different schemas.
```

Comment:
This is a nice refactor

Reply:
The Boy Scout Rule: "Leave the campground cleaner than you found it" 🤣